File: .claude/worktrees/agent-a3568cce/FoodDetectionApp/AnalyticsService.swift
import Foundation
import FirebaseAnalytics

enum AnalyticsService {
    static func logFoodScanned(dish: String, source: String) {
        Analytics.logEvent("food_scanned", parameters: [
            "dish": dish,
            "source": source,
        ])
    }

    static func logFoodLogged(dish: String, calories: Int) {
        Analytics.logEvent("food_logged", parameters: [
            "dish": dish,
            "calories": calories,
        ])
    }

    static func logManualSearch(query: String) {
        Analytics.logEvent("manual_search", parameters: [
            "query": query,
        ])
    }

    static func logCoachQuery(query: String) {
        Analytics.logEvent("coach_query", parameters: [
            "query": query,
        ])
    }

    static func logScaleConnected() {
        Analytics.logEvent("scale_connected", parameters: nil)
    }

    static func logOnboardingComplete() {
        Analytics.logEvent("onboarding_complete", parameters: nil)
    }

    static func logWaterLogged(ml: Int) {
        Analytics.logEvent("water_logged", parameters: [
            "ml": ml,
        ])
    }
}

File: .claude/worktrees/agent-ae6253e0/android/app/src/main/java/com/foodsense/android/services/AnalyticsService.kt
package com.foodsense.android.services

import android.os.Bundle
import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.analytics.ktx.analytics
import com.google.firebase.ktx.Firebase

object AnalyticsService {
    private val analytics: FirebaseAnalytics by lazy { Firebase.analytics }

    fun logFoodScanned(dish: String, source: String) {
        analytics.logEvent("food_scanned", Bundle().apply {
            putString("dish", dish)
            putString("source", source)
        })
    }

    fun logFoodLogged(dish: String, calories: Int) {
        analytics.logEvent("food_logged", Bundle().apply {
            putString("dish", dish)
            putInt("calories", calories)
        })
    }

    fun logManualSearch(query: String) {
        analytics.logEvent("manual_search", Bundle().apply {
            putString("query", query)
        })
    }

    fun logCoachQuery(query: String) {
        analytics.logEvent("coach_query", Bundle().apply {
            putString("query", query)
        })
    }

    fun logScaleConnected() {
        analytics.logEvent("scale_connected", null)
    }

    fun logOnboardingComplete() {
        analytics.logEvent("onboarding_complete", null)
    }

    fun logWaterLogged(ml: Int) {
        analytics.logEvent("water_logged", Bundle().apply {
            putInt("ml", ml)
        })
    }
}
